File: server/Models/Vault.cs
using System.ComponentModel.DataAnnotations;

namespace keepr.Models;

public class Vault : RepoItem<int>
{
  [MaxLength(255)] public string Name { get; set; }
  [MaxLength(1000)] public string Description { get; set; }
  [Url, MaxLength(1000)] public string Img { get; set; }
  public bool? IsPrivate { get; set; }
  [MaxLength(255)] public string CreatorId { get; set; }
  public Profile Creator { get; set; }
}

File: client/src/models/Vault.js
import { APIItem } from "./APIItem.js";
import { Profile } from "./Profile.js";

export class Vault extends APIItem{
  constructor(data){
    super(data)
    this.name= data.name
    this.description= data.description
    this.img= data.img
    this.isPrivate= data.isPrivate
    this.creatorId= data.creatorId
this.creator = new Profile(data.creator)
  }
}